src/test/lib/storage/segment_access_counter_test.cpp

```diff
@@ -196,4 +196,46 @@ TEST_F(SegmentAccessCounterTest, AccessPattern8) {
   EXPECT_EQ(_access_pattern(positions), AccessPattern::Random);
 }
 
+TEST_F(SegmentAccessCounterTest, SelfAssignment) {
+  SegmentAccessCounter counter;
+  counter[SegmentAccessCounter::AccessType::Point] = 10;
+  counter[SegmentAccessCounter::AccessType::Sequential] = 20;
+
+  // Store original values to check after self-assignment
+  const auto original_point = counter[SegmentAccessCounter::AccessType::Point].load();
+  const auto original_sequential = counter[SegmentAccessCounter::AccessType::Sequential].load();
+
+  // Assign via a reference to avoid -Wself-assign-overloaded
+  SegmentAccessCounter& ref = counter;
+  counter = ref;
+
+  EXPECT_EQ(counter[SegmentAccessCounter::AccessType::Point], original_point);
+  EXPECT_EQ(counter[SegmentAccessCounter::AccessType::Sequential], original_sequential);
+}
+
+TEST_F(SegmentAccessCounterTest, InequalityOperator) {
+  SegmentAccessCounter counter1;
+  SegmentAccessCounter counter2;
```

Comment:
auto

Reply:
Maybe change a all counters to not have the majority zero-initialized in this test for equality and compare to all possible counters differing in one access type for inequality? Thus, we would actually notice if the comparison operators miss on some comparisons.